src/radical/pilot/controller/input_file_transfer_worker.py

```diff
@@ -174,22 +171,16 @@ def run(self):
                                 logger.debug(log_msg)
 
                                 # Execute the transfer.
-                                logger.debug ("saga.fs.File ('%s')" % input_file_url)
-                                input_file = saga.filesystem.File(
-                                    input_file_url,
-                                    session=self._session
-                                )
-
                                 if CREATE_PARENTS in sd['flags']:
                                     copy_flags = saga.filesystem.CREATE_PARENTS
                                 else:
                                     copy_flags = 0
 
-                                try :
-                                    input_file.copy(target, flags=copy_flags)
-                                except Exception as e :
-                                    logger.exception (e)
-                                input_file.close()
+                                try:
+                                    saga_dir.copy(input_file_url, target, flags=copy_flags)
```

Comment:
input staging now is done on the saga_dir object -- shouldn't the same be applied to the output staging, or is that not needed there / already there differently?